File: lecture_notes/08 collections/a-union.py
def union_cities(s1, s2):
    difference = set(s2) - set(s1)
    result = s1 + list(difference)
    return result


cities1 = { 'Beijing', 'Kiev' }
cities2 = { 'Kiev', 'London', 'Baghdad' }

print(union_cities(cities1, cities2))

File: lecture_notes/08 collections/a-union.js
'use strict';

const union = (s1, s2) => {
  const ds = s1.slice(0);
  for (let i = 0; i < s2.length; i++) {
    const item = s2[i];
    if (!ds.includes(item)) ds.push(item);
  }
  return ds;
};

// const union = (s1, s2) => new Set([...s1, ...s2]);

// Usage

const cities1 = ['Beijing', 'Kiev'];
const cities2 = ['Kiev', 'London', 'Baghdad'];
console.dir({ cities1, cities2 });

const results = union(cities1, cities2);
console.dir(results);
